cli/commands/hcl/format/format.go

```diff
@@ -63,16 +64,16 @@ func Run(opts *options.TerragruntOptions) error {
 		skipFile := false
 		// Ignore any files that are in the cache or scaffold dir
 		pathList := strings.Split(fname, "/")
-		if util.ListContainsElement(pathList, util.TerragruntCacheDir) {
+		if slices.Contains(pathList, util.TerragruntCacheDir) {
```

Comment:
I think we should exclude `stackDir = ".terragrunt-stack"` too, most probably should be defined a list of excluded directories to not grow multiple ifs

Reply:
I think we should address that logic for `.terragrunt-stack` in this PR, as it's a bug once stacks are GA otherwise.

We can look to pull in usage of the `discovery` package later, which has a `hidden` setting for avoiding discovery of hidden configurations.

Reply:
Added exclude list.